hw2/smart_args/smart_args_decorator.py

```diff
@@ -0,0 +1,66 @@
+from functools import update_wrapper
+from inspect import getfullargspec, FullArgSpec
+from typing import TypeVar, Generic, Callable, List, Tuple
+
+from hw2.smart_args.no_value import NoValue
+from hw2.smart_args.supported_magic_args import MagicArgument
+
+
+class MagicArgumentsMisuseError(TypeError):
```

Comment:
Should we create a new error class? You can use a standard class with a customized message